subsys/bluetooth/host/l2cap.c

```diff
@@ -2265,6 +2266,12 @@ static void l2cap_chan_send_credits(struct bt_l2cap_le_chan *chan,
 		credits = chan->rx.init_credits;
 	}
 
+	/* Don't send back more than the initial amount. */
```

Comment:
Why? Is it not allowed by spec? This looks to me like a band-aid solution for some other problem in the stack. Why does it happen that we try to `l2cap_chan_send_credits` more credits than we should?

Reply:
Think the stack has this assumption, even if it's not in writing anywhere.

Reply:
yeah initially, the fix was on the caller side. I decided to put it here to gate it at the very last point, just in case I missed something in one of the other calls. 
We can revisit later.

Reply:
Isn't the number of credits we can give limited by the number of buffers available? We don't get any more buffers than the initial amount, so this change makes sense to me. But I may be misunderstanding something.

Reply:
yes exactly, this is the assumption. In the zephyr stack, the credits are an image of the number of the receive buffers. If they get out of sync, you're in trouble.

I think in nordic's softdevice, it was on the application to send the credits (public api), so it allowed the app to process the buffers before deciding to accept more. it also allowed user error if they sent too much credits. I'm not sure but maybe the number of buffers were runtime-configurable, hence that API.

As I said, i think in zephyr that doesn't make sense, as the number of buffers is fixed at compile time.

Reply:
So we are the losing sync between the number of free buffers and the credits we send?~ Does the limit added here prevent the desync? Or is the sync already lost, and we are patching the symptom; that is some code calling `l2cap_chan_send_credits` with more credits than it is possible to have?

Reply:
yeah this code prevents the desync. There is some code callign l2cap_chan_send_credits with more credits than it is possible to have, yes. I figured I'd have the check in one place rather than two (or just the orig. caller and possibly leave another bug in if I didn't gate it here).
I can rummage thru my commits to find out where I originally put the limit if you so prefer. But i want to leave that one too.